gzhttp/compress.go

```diff
@@ -112,7 +123,7 @@ func (w *GzipResponseWriter) Write(b []byte) (int, error) {
 		ct := hdr.Get(contentType)
 		if cl == 0 || cl >= w.minSize && (ct == "" || w.contentTypeFilter(ct)) {
 			// If the current buffer is less than minSize and a Content-Length isn't set, then wait until we have more data.
-			if len(w.buf) < w.minSize && cl == 0 {
+			if len(w.buf) < w.minSize && cl == 0 || (w.jitterBuffer > 0 && len(w.buf) < w.jitterBuffer) {
```

Comment:
Why is the insertion of the content type header conditional on the entropy available for the jitter ?

Responses of size smaller than `jitterBuffer` never seem to actually get a `Content-Type` header in my tests. As a note, I am quite sure that the underlying handlers are not guaranteed to call `Flush`, so I don't see how the `Content-Type` would be set.

Reply:
@adriansmares Please open an issue with a reproducer.